agent/engine/docker_container_engine_test.go

```diff
@@ -717,3 +717,123 @@ func TestUnavailableVersionError(t *testing.T) {
 		t.Fatal("Error was not a named error")
 	}
 }
+
+func TestStatsNormalExit(t *testing.T) {
+	mockDocker, client, _, done := dockerclientSetup(t)
+	defer done()
+	time1 := time.Now()
+	time2 := time1.Add(1 * time.Second)
+	mockDocker.EXPECT().Stats(gomock.Any()).Do(func(x interface{}) {
+		opts := x.(docker.StatsOptions)
+		defer close(opts.Stats)
+		if opts.ID != "foo" {
+			t.Fatalf("Expected ID foo, got %s", opts.ID)
+		}
+		if opts.Stream != true {
+			t.Fatal("Expected stream to be true")
+		}
+		opts.Stats <- &docker.Stats{
+			Read: time1,
+		}
+		opts.Stats <- &docker.Stats{
+			Read: time2,
+		}
+	})
+	ctx := context.TODO()
+	stats, err := client.Stats("foo", ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	stat := <-stats
+	checkStatRead(t, stat, time1)
+	stat = <-stats
+	checkStatRead(t, stat, time2)
+	stat = <-stats
+	if stat != nil {
```

Comment:
No, the assertion is different.  The first two reads should have valid data, the last one should be `nil` because the channel is closed (maybe I should change it to `_, ok <-stats` and check `if ok`?).